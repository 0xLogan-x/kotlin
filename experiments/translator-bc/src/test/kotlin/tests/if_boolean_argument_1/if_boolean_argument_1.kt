fun if_boolean_argument_1(value: Boolean): Int {
    return if (value) 11 else 411
}