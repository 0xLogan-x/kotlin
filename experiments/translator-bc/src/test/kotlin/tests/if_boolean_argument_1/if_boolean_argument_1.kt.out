@exceptions.str.1 = private unnamed_addr constant  [59 x i8] c"Exception in thread main kotlin.KotlinNullPointerException\00", align 1
declare void @llvm.memcpy.p0i8.p0i8.i64(i8* nocapture, i8* nocapture readonly, i64, i32, i1)
declare i8* @malloc_heap(i32)
declare i32 @printf(i8*, ...)
declare void @abort()
%class.Nothing = type { }
define weak i32 @if_boolean_argument_1_Boolean(i1  %if_boolean_argument_1.value) #0
{
%if_boolean_argument_1.value.addr = alloca i1, align 4
store i1 %if_boolean_argument_1.value, i1* %if_boolean_argument_1.value.addr, align 4
%var2 = load i1* %if_boolean_argument_1.value.addr, align 4
%var3 = alloca i32, align 4
br i1 %var2, label %label.if_then.1, label %label.if_else.2
label.if_then.1:
store i32 11, i32* %var3, align 4
br label %label.if_end.3
label.if_else.2:
store i32 411, i32* %var3, align 4
br label %label.if_end.3
label.if_end.3:
%var4 = load i32* %var3, align 4
ret i32 %var4
unreachable
}

